src/statcache.c

```diff
@@ -426,6 +433,70 @@ void stat_cache_value_set(stat_cache_t *cache, const char *path, struct stat_cac
     return;
 }
 
+// A negative entry is an item in the cache which represents a miss,
+// so we can cache its non-existence and regulate how often
+// we make a propfind request to the server to check if it has
+// come into existence.
+bool stat_cache_is_negative_entry(struct stat_cache_value value) {
+    // The struct stat st gets zero'ed out when we put a negative entry in the cache.
+    // If an extant item is put in the cache, at st_mode will be non-zero.
+    // So use st_mode as our check for non-existence
+    if (value.st.st_mode == 0) return true;
+    else return false;
+}
+
+
+// Create or update a negative entry in the stat cache for a deleted or non-existent object
```

Comment:
This roughly replaces stat_cache_delete, which I have left in in case we need to do true deletes. A 'delete' is now just the insertion of a negative entry.